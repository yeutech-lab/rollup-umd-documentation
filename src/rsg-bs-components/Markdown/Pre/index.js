export { default } from 'rsg-components/Markdown/Pre/PreRenderer';
