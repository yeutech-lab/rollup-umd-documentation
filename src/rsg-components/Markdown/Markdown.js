import React from 'react'; // eslint-disable-line no-unused-vars
import PropTypes from 'prop-types';
import { compiler } from 'markdown-to-jsx';
import Table from 'bootstrap-styled/lib/Table'; // eslint-disable-line no-unused-vars
import Thead from 'bootstrap-styled/lib/Table/Thead'; // eslint-disable-line no-unused-vars
import Tbody from 'bootstrap-styled/lib/Table/Tbody'; // eslint-disable-line no-unused-vars
import Th from 'bootstrap-styled/lib/Table/Th'; // eslint-disable-line no-unused-vars
import Tr from 'bootstrap-styled/lib/Table/Tr'; // eslint-disable-line no-unused-vars
import Td from 'bootstrap-styled/lib/Table/Td'; // eslint-disable-line no-unused-vars
import Link from '../Link'; // eslint-disable-line no-unused-vars
import Text from '../Text'; // eslint-disable-line no-unused-vars
import Para from '../Para'; // eslint-disable-line no-unused-vars
import Code from '../Code'; // eslint-disable-line no-unused-vars
import MarkdownHeading from './MarkdownHeading/MarkdownHeadingRenderer'; // eslint-disable-line no-unused-vars
import List from './List/ListRenderer'; // eslint-disable-line no-unused-vars
import Blockquote from './Blockquote/BlockquoteRenderer'; // eslint-disable-line no-unused-vars
import Pre from './Pre/PreRenderer'; // eslint-disable-line no-unused-vars
import Checkbox from './Checkbox/CheckboxRenderer'; // eslint-disable-line no-unused-vars
import Hr from './Hr/HrRenderer'; // eslint-disable-line no-unused-vars


export const baseOverrides = {
  a: {
    component: Link,
  },
  h1: {
    component: MarkdownHeading,
    props: {
      level: 1,
    },
  },
  h2: {
    component: MarkdownHeading,
    props: {
      level: 2,
    },
  },
  h3: {
    component: MarkdownHeading,
    props: {
      level: 3,
    },
  },
  h4: {
    component: MarkdownHeading,
    props: {
      level: 4,
    },
  },
  h5: {
    component: MarkdownHeading,
    props: {
      level: 5,
    },
  },
  h6: {
    component: MarkdownHeading,
    props: {
      level: 6,
    },
  },
  p: {
    component: Para,
    props: {
      semantic: 'p',
    },
  },
  em: {
    component: Text,
    props: {
      semantic: 'em',
    },
  },
  strong: {
    component: Text,
    props: {
      semantic: 'strong',
    },
  },
  ul: {
    component: List,
  },
  ol: {
    component: List,
    props: {
      ordered: true,
    },
  },
  blockquote: {
    component: Blockquote,
  },
  code: {
    component: Code,
  },
  pre: {
    component: Pre,
  },
  input: {
    component: Checkbox,
  },
  hr: {
    component: Hr,
  },
  table: {
    component: Table,
  },
  thead: {
    component: Thead,
  },
  th: {
    component: Th,
    props: {
      header: true,
    },
  },
  tbody: {
    component: Tbody,
  },
  tr: {
    component: Tr,
  },
  td: {
    component: Td,
  },
};

export const inlineOverrides = {
  ...baseOverrides,
  p: {
    component: Text,
  },
};

function Markdown({ text, inline }) {
  const overrides = inline ? inlineOverrides : baseOverrides;
  return compiler(text, { overrides, forceBlock: true });
}

Markdown.propTypes = {
  text: PropTypes.string.isRequired,
  inline: PropTypes.bool,
};

export default Markdown;
