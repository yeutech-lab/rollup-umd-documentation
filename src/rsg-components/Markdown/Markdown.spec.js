import React from 'react';
import { html } from 'cheerio';
import Markdown from './Markdown';

describe('Markdown', () => {
	const expectSnapshotToMatch = markdown => {
		const actual = render(<Markdown text={markdown} />);

		expect(html(actual)).toMatchSnapshot();
	};

	it('should forward DOM attributes onto resulting HTML', () => {
		const markdown =
			'<a href="test.com" id="preserve-my-id" class="preserve-my-class">Something</a>';

		const actual = mount(<Markdown text={markdown} />);

		expect(actual.find('a').props().id).toEqual('preserve-my-id');
		expect(actual.find('a').props().className).toContain('preserve-my-class');
	});

	it('should render links', () => {
		expectSnapshotToMatch('a [link](http://test.com)');
	});

	it('should render headings', () => {
		expectSnapshotToMatch(`
# one
## two
### three
#### four
##### five
###### six
`);
	});

	it('should render paragraphs', () => {
		expectSnapshotToMatch(`
a paragraph

another paragraph
		`);
	});

	it('should render emphasis and strong text', () => {
		expectSnapshotToMatch(`
this text is **strong**

and this is _emphasized_
		`);
	});

	it('should render unordered lists', () => {
		expectSnapshotToMatch(`
* list
* item
* three
`);
	});

	it('should render ordered lists', () => {
		expectSnapshotToMatch(`
1. list
1. item
1. three
`);
	});

	it('should render mixed nested lists', () => {
		expectSnapshotToMatch(`
* list 1
* list 2
  1. Sub-list
  1. Sub-list
  1. Sub-list
* list 3
`);
	});

	it('should render check-lists', () => {
		expectSnapshotToMatch(`
* [ ] to do 1
* [ ] to do 2
* [x] to do 3
`);
	});

	it('should render a blockquote', () => {
		expectSnapshotToMatch(`
> This is a blockquote.
> And this is a second line.
`);
	});

	it('should render pre-formatted text', () => {
		expectSnapshotToMatch(`
    this is preformatted
    so is this
`);
	});

	it('should render code blocks without escaping', () => {
		expectSnapshotToMatch(`
\`\`\`html
<foo></foo>
\`\`\`
`);
	});

	it('should render inline code with escaping', () => {
		expectSnapshotToMatch('Foo `<bar>` baz');
	});

	it('should render a horizontal rule', () => {
		expectSnapshotToMatch(`---`);
	});

	it('should render a table', () => {
		expectSnapshotToMatch(`
| heading 1 | heading 2 |
| --------- | --------- |
| foo		| bar		|
| more foo	| more bar	|
`);
	});
});

describe('Markdown inline', () => {
	const expectSnapshotToMatch = markdown => {
		const actual = render(<Markdown text={markdown} inline />);

		expect(html(actual)).toMatchSnapshot();
	};

	it('should render text in a span', () => {
		expectSnapshotToMatch('Hello world!');
	});
});
