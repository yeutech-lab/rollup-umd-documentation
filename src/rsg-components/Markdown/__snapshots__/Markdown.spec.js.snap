// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown inline should render text in a span 1`] = `

<span class="rsg--text-11 rsg--inheritSize-12 rsg--baseColor-16">
  Hello world!
</span>

`;

exports[`Markdown should render a blockquote 1`] = `

<blockquote class="rsg--blockquote-25">
  <p class="rsg--para-2">
    This is a blockquote.
And this is a second line.
  </p>
</blockquote>

`;

exports[`Markdown should render a horizontal rule 1`] = `

<hr class="rsg--hr-28">

`;

exports[`Markdown should render a table 1`] = `

<table class="rsg--table-29">
  <thead class="rsg--thead-30">
    <tr>
      <th class="rsg--th-32 rsg--td-31">
        heading 1
      </th>
      <th class="rsg--th-32 rsg--td-31">
        heading 2
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td class="rsg--td-31">
        foo
      </td>
      <td class="rsg--td-31">
        bar
      </td>
    </tr>
    <tr>
      <td class="rsg--td-31">
        more foo
      </td>
      <td class="rsg--td-31">
        more bar
      </td>
    </tr>
  </tbody>
</table>

`;

exports[`Markdown should render check-lists 1`] = `

<ul class="rsg--list-21">
  <li class="rsg--li-23">
    <input type="checkbox"
           readonly
           class="rsg--input-24"
    >
    to do 1
  </li>
  <li class="rsg--li-23">
    <input type="checkbox"
           readonly
           class="rsg--input-24"
    >
    to do 2
  </li>
  <li class="rsg--li-23">
    <input type="checkbox"
           checked
           readonly
           class="rsg--input-24"
    >
    to do 3
  </li>
</ul>

`;

exports[`Markdown should render code blocks without escaping 1`] = `

<pre class="rsg--pre-26">
  <code class="lang-html rsg--code-27">
    <foo>
    </foo>
  </code>
</pre>

`;

exports[`Markdown should render emphasis and strong text 1`] = `

<div>
  <p class="rsg--para-2">
    this text is
    <strong class="rsg--text-11 rsg--inheritSize-12 rsg--baseColor-16 rsg--strong-19">
      strong
    </strong>
  </p>
  <p class="rsg--para-2">
    and this is
    <em class="rsg--text-11 rsg--inheritSize-12 rsg--baseColor-16 rsg--em-18">
      emphasized
    </em>
  </p>
</div>

`;

exports[`Markdown should render headings 1`] = `

<div>
  <div class="rsg--spacing-3">
    <h1 class="rsg--heading-4 rsg--heading1-5">
      one
    </h1>
  </div>
  <div class="rsg--spacing-3">
    <h2 class="rsg--heading-4 rsg--heading2-6">
      two
    </h2>
  </div>
  <div class="rsg--spacing-3">
    <h3 class="rsg--heading-4 rsg--heading3-7">
      three
    </h3>
  </div>
  <div class="rsg--spacing-3">
    <h4 class="rsg--heading-4 rsg--heading4-8">
      four
    </h4>
  </div>
  <div class="rsg--spacing-3">
    <h5 class="rsg--heading-4 rsg--heading5-9">
      five
    </h5>
  </div>
  <div class="rsg--spacing-3">
    <h6 class="rsg--heading-4 rsg--heading6-10">
      six
    </h6>
  </div>
</div>

`;

exports[`Markdown should render inline code with escaping 1`] = `

<p class="rsg--para-2">
  Foo
  <code class="rsg--code-27">
    &lt;bar&gt;
  </code>
  baz
</p>

`;

exports[`Markdown should render links 1`] = `

<p class="rsg--para-2">
  a
  <a href="http://test.com"
     class="rsg--link-0"
  >
    link
  </a>
</p>

`;

exports[`Markdown should render mixed nested lists 1`] = `

<ul class="rsg--list-21">
  <li class="rsg--li-23">
    list 1
  </li>
  <li class="rsg--li-23">
    list 2
    <ol class="rsg--list-21 rsg--ordered-22">
      <li class="rsg--li-23">
        Sub-list
      </li>
      <li class="rsg--li-23">
        Sub-list
      </li>
      <li class="rsg--li-23">
        Sub-list
      </li>
    </ol>
  </li>
  <li class="rsg--li-23">
    list 3
  </li>
</ul>

`;

exports[`Markdown should render ordered lists 1`] = `

<ol class="rsg--list-21 rsg--ordered-22">
  <li class="rsg--li-23">
    list
  </li>
  <li class="rsg--li-23">
    item
  </li>
  <li class="rsg--li-23">
    three
  </li>
</ol>

`;

exports[`Markdown should render paragraphs 1`] = `

<div>
  <p class="rsg--para-2">
    a paragraph
  </p>
  <p class="rsg--para-2">
    another paragraph
  </p>
</div>

`;

exports[`Markdown should render pre-formatted text 1`] = `

<pre class="rsg--pre-26">
  <code class="rsg--code-27">
    this is preformatted
so is this
  </code>
</pre>

`;

exports[`Markdown should render unordered lists 1`] = `

<ul class="rsg--list-21">
  <li class="rsg--li-23">
    list
  </li>
  <li class="rsg--li-23">
    item
  </li>
  <li class="rsg--li-23">
    three
  </li>
</ul>

`;
